Doctor home improvements

Thommond opened the pull request:
Created the doctorHome.php file first set of forms which allow doctors to filter then see all previous appointments which match their filter results.

Updated the docappt.php, register.php so that the file contents would match the database and allow for workflow on doctorHome.php to continue. Including the 105 variable error.

Updated data.sql and schema.sql with new database changes like adding day foreign key to the prescriptions table and adding more mock data examples so doctorHome.php could be properly tested. 

Updated the doctorHome.php file so that after filtering previous results of appointments doctors can then look at all upcoming appointments from today until the selected date of the doctor's choice.



Finally, Updated styles.css with minor updates like making links more readable in tables and centering of data.

Thommond committed:
Displaying unique doctor name.

Created Pseudo code guildlines.

Thommond committed:
Created forms for search option on doctorHome.php

Thommond committed:
Updated data.sql to have mock data for doctorhome.php

Updated schema.sql so less merge conflicts will appear later.

Updated doctorHome.php so that doctors can filter results of previous 
appointments.

Thommond committed:
Updated the doctorHome.php to handle one field at a time for filtration.

Updated so that it shows the user either a table of result rows or a 
message saying no rows with filter.

Thommond committed:
Added mock data for appointments to get mutiple displayed to front-end.

Updated doctorHome.php with some Todo comments and a few minor tweaks.

Thommond committed:
Create Mock data for prescriptions, appointments and more user mock for the need of docotorHome.php

Thommond committed:
Added more testing of possible errors in doctorHome.php.

Added more mock data to data.sql.

Thommond committed:
Updated prescriptions table to have foriegn key of date from appointments.

Updated data.sql to have more mock data related to docHome.

Upated docappt.php so when their is no results only displays error.

Thommond committed:
Updated docapp.php and roster.php from the_date to day in database retreival.

Updated doctorHome.php so it shows all dates after the date selected in 
second form.

Thommond committed:
Updated styles so that links in the table will be able to be seen.

Updated doctorHome.php to have place holder links for 
patient_of_doctor.php so to be file.

Thommond committed:
Merge branch 'main' into doctorHome-improvements

BrendanHorst commented on views/admin/docappt.php:
```diff
@@ -43,9 +43,11 @@
 
+        $day = $_POST['day'];
         $patient = $_POST['patient_id'];
         $day = $_POST['day'];
```
day is set twice

BrendanHorst commented on views/admin/docappt.php:
```diff
@@ -101,6 +106,8 @@
+          if ($id and $id != 0) {
+
           $sql = "SELECT Fname, Lname, id FROM users
                   WHERE id LIKE '$id'";
```
the block inside this `if` should be indented

BrendanHorst commented on views/admin/docappt.php:
```diff
@@ -123,18 +130,23 @@
+           echo "<label for='patients_name' class='display_only'>Patient Name:";
+             echo "<p class='display_only' >" . $patient_name . "</p>";
+           echo "</label>";
+         }
```
These `echo`s are not indented uniformly, and this is part of the block that needs indenting, so the closing bracket should be on  the same indentation level as the `if` that it closes

BrendanHorst commented on views/admin/docappt.php:
```diff
@@ -123,18 +130,23 @@
+          $day = $_POST['day'];
           $patient = $_POST['patient'];
           $id = $_POST['doc'];
           $day = $_POST['date'];
```
day is defined twice

BrendanHorst commented on views/admin/docappt.php:
```diff
@@ -123,18 +130,23 @@
+
+
+        }
+        ?>
 
       <?php
 
         if(isset($_POST['press'])) {
```
It seems a little redundant to close the `<?php>` only to open it again right away.  Unless you were planning on adding html here I think it be better to combine them

BrendanHorst commented on views/home/doctorHome.php:
```diff
@@ -27,11 +27,274 @@
+        $Amed = $_POST['amed'];
+        $Nmed = $_POST['nmed'];
+
+        $day = 'day';
+        $Last = 'Lname';
+        $first = 'Fname';
+        $com = 'Comment';
+        $morn = 'morning_med';
+        $aft = 'afternoon_med';
+        $night = 'night_med';
```
these variables seem redundant

BrendanHorst commented on views/home/doctorHome.php:
```diff
@@ -27,11 +27,274 @@
+
+                  }
+
+                  else {
+
+                    if (!empty($Amed)) {
+
+                      $filter = $Amed . '%';
+                      $column = $aft;
+
+                    }
+
+                    else {
+
+                      if (!empty($Nmed)) {
```
this whole section isn't wrong, but just a reminder that `elseif` does exist and would save a lot of lines and indentations

BrendanHorst commented on views/home/doctorHome.php:
```diff
@@ -27,11 +27,274 @@
+           die("<p class='error'>The field you entered has no results</p>");
+         }
+
+         else {
+
+           // If result is not empty then display table
+         echo '<table>';
```
The block inside this `else` is not indented

BrendanHorst commented on views/home/doctorHome.php:
```diff
@@ -27,11 +27,274 @@
+         echo '<th>Patient Page</th>';
+         echo '</tr>';
+
+         if ($result->fetch_assoc()) {
+
+            // Every result row displayed in table if there is any
+           while ($row = $result->fetch_assoc()) {
```
I'm not confidant on this, but wouldn't running `fetch_assoc()` in the `if` cause the first record in row to be excluded?

BrendanHorst commented on views/home/doctorHome.php:
```diff
@@ -27,11 +27,274 @@
+             echo '<td>' . $row['comment'] . '</td>';
+             echo '<td>' . $row['morning_med'] . '</td>';
+             echo '<td>' . $row['afternoon_med'] . '</td>';
+             echo '<td>' . $row['night_med'] . '</td>';
+            echo '<td>' . "<a class='table_link' href='#'>Patient Page</a>" . '</td>';
+             echo '</tr>';
```
incorrect indentation

BrendanHorst reviewed (changes requested):
Looks alright, I mostly noticed indentation errors and redundant lines

Thommond committed:
Updated fixs of indentation in doctorHome.php and docappt.php

Updated long nested else and ifs to else ifs respectfully. 

Updated so redundant variables where not needed in doctorHome.php.

Thommond committed:
Updated data.sql to fit needs of doctorHome.php

Udpated login.php to have indentation and todo comment.

Updated minor errors in doctorHome.php

Thommond committed:
Update the query in doctorHome.php

Thommond committed:
Merge branch 'doctorHome-improvements' of github.com:Thommond/Retire-Inspired into doctorHome-improvements

Thommond committed:
Updated day reduncany on docappt.php

Thommond committed:
Updated forms on docappt.php
